### PR TITLE
fix(ci): fix tap test failure reporting

### PR DESCRIPTION
**Pull request by InteractiveTimmy**
# Pull Request Template

## Description

The scope of this pull request is to address the tap tests not reporting failures when failures are present.

Fixes # [SPARK-133294](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-133294)

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by taymoork2** [comment]
The testing command runs a for loop so its not going to exit like you'd want
It's gonna exit 0 because the loop finished executing all the spawn commands from all the packages that it was given (regardless if its 1 or 10)

Its more work to grab the exit code from tooling + run more tests if more packages are passed than it is to just look in the xml after the fact

**Comment by lalli-flores** [comment]
Alright, thanks for the explanation.

File: .circleci/config.yml [hunk 1]
```diff
@@ -337,6 +337,20 @@ jobs:
...
+      - run:
+          name: 'Check if tap test failed'
+          command: |
+            mkdir -p reports/junit/karma
+            FAILURES=$(find reports/junit/karma/ -iname '*.xml' -print0 | xargs -0 sed -n 's/.*failures=\"\([^\"]*\).*/\1/p' | xargs)
+
+            echo Failures $FAILURES
+
+            if [[ $FAILURES =~ ^([0 ])*$ ]]; then
+              echo 'Tap test succeeded'
```
Comment (adamweeks) [inline .circleci/config.yml:349]: Would be nice to have the result have the package it was testing echoed as well:

```suggestion
              echo 'Tap test for << parameters.package >> succeeded'
```

Comment (InteractiveTimmy) [reply .circleci/config.yml]: I added this to the commands being run, please let me know if these changes look okay.

**Pull request merged**